diff --git a/target/linux/rockchip/image/armv8.mk b/target/linux/rockchip/image/armv8.mk
index 5ca6bf7..1a503f3 100644
--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -285,6 +285,16 @@ define Device/friendlyarm_nanopi-r76s
 endef
 TARGET_DEVICES += friendlyarm_nanopi-r76s
 
+define Device/hinlink_h28k
+  $(Device/rk3528)
+  DEVICE_VENDOR := HINLINK
+  DEVICE_MODEL := H28K
+  DEVICE_DTS := rk3528-hinlink-h28k
+  UBOOT_DEVICE_NAME := hinlink-h28k-rk3528
+  DEVICE_PACKAGES := kmod-input-adc-keys kmod-saradc-rockchip kmod-r8169
+endef
+TARGET_DEVICES += hinlink_h28k
+
 define Device/huake_guangmiao-g4c
   $(Device/rk3399)
   DEVICE_VENDOR := Huake-Cloud
