diff --git a/target/linux/rockchip/armv8/base-files/etc/init.d/phy-leds b/target/linux/rockchip/armv8/base-files/etc/init.d/phy-leds
index 90767cbbd4b6..4e135479ef2a 100755
--- a/target/linux/rockchip/armv8/base-files/etc/init.d/phy-leds
+++ b/target/linux/rockchip/armv8/base-files/etc/init.d/phy-leds
@@ -74,6 +74,12 @@ boot() {
 		netled_set "stmmac-0:01:amber:wan" "eth0" "link_10 link_100 link_1000" "0" "rx tx" "1"
 		netled_set "stmmac-0:01:green:wan" "eth0" "link_10 link_100 link_1000" "1" "rx tx" "0"
 		;;
+	hinlink,h28k)
+		netled_set "enp1s0-1::lan" "eth1" "link_10 link_100 link_1000" "1" "rx tx" "0"
+		netled_set "enp1s0-2::lan" "eth1" "link_10 link_100 link_1000" "0" "rx tx" "1"
+		netled_set "stmmac-0:01:green:lan" "eth0" "link_10 link_100 link_1000" "1" "rx tx" "0"
+		netled_set "stmmac-0:01:amber:lan" "eth0" "link_10 link_100 link_1000" "0" "rx tx" "1"
+		;;
 	friendlyarm,nanopi-r4s)
 		netled_set "enp1s0-1::lan" "" "link_10 link_100 link_1000" "1" "rx tx" "0"
 		netled_set "enp1s0-2::lan" "" "link_10 link_100 link_1000" "0" "rx tx" "1"
