diff --git a/target/linux/rockchip/image/armv8.mk b/target/linux/rockchip/image/armv8.mk
index f322d35fb0..1d014eb850 100644
--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -224,6 +224,18 @@ define Device/friendlyarm_nanopi-r6s
 endef
 TARGET_DEVICES += friendlyarm_nanopi-r6s
 
+define Device/hinlink_h28k
+  DEVICE_VENDOR := HINLINK
+  DEVICE_MODEL := H28K
+  SOC := rk3528
+  KERNEL_LOADADDR := 0x03000000
+  DEVICE_DTS := rockchip/rk3528-hinlink-h28k
+  UBOOT_DEVICE_NAME := generic-rk3528
+  BOOT_FLOW := pine64-img
+  DEVICE_PACKAGES := kmod-input-adc-keys kmod-saradc-rockchip kmod-r8169
+endef
+TARGET_DEVICES += hinlink_h28k
+
 define Device/huake_guangmiao-g4c
   DEVICE_VENDOR := Huake-Cloud
   DEVICE_MODEL := GuangMiao G4C
